src/NHibernate/Mapping/Table.cs

```diff
@@ -414,9 +414,12 @@ public string SqlCreateString(Dialect.Dialect dialect, IMapping p, string defaul
 				buf.Append(StringHelper.CommaSpace).Append(PrimaryKey.SqlConstraintString(dialect, defaultSchema));
 			}
 
-			foreach (UniqueKey uk in UniqueKeyIterator)
+			if (dialect.SupportsNotNullUnique)
```

Comment:
I think it's more accurate way is to check that `uk.SqlConstraintString(dialect)` returns not empty string before adding to `buf`